src/monodroid/jni/dylib-mono.h

```diff
@@ -442,6 +446,10 @@ class DylibMono
 	typedef void                   (*monodroid_mono_install_assembly_refonly_preload_hook_fptr) (MonoAssemblyPreLoadFunc func, void *user_data);
 	typedef int                    (*monodroid_mono_runtime_set_main_args_fptr) (int argc, char* argv[]);
 	typedef void                   (*mono_aot_register_module_fptr) (void* aot_info);
+	typedef MonoProfilerHandle     (*monodroid_mono_profiler_create_fptr) (MonoProfiler* profiler);
+	typedef void                   (*monodroid_mono_profiler_set_jit_done_callback_fptr) (MonoProfilerHandle handle, void *done_ftn);
+	typedef void                   (*monodroid_mono_profiler_set_thread_started_callback_fptr) (MonoProfilerHandle handle, void *start_ftn);
```

Comment:
I think you are mixing the `thread_started` and `jit_done` events here. The `jit_done` has correct signature:

    static void
    jit_done (MonoProfiler *prof, MonoMethod *method, MonoJitInfo* jinfo)

as the event is defined like this:

    MONO_PROFILER_EVENT_2(jit_done, JitDone, MonoMethod *, method, MonoJitInfo *, jinfo)

> Should we provide a better prototype for start_ftn and done_ftn than void*?

It is good to be on the safe side. Opened https://github.com/xamarin/xamarin-android/pull/2675 for it and `jit_done` as well.

> Part of the idea behind the C++ migration is to improve type safety. Let's not lose it by using void* everywhere.

Well, it is not the case in many places. The original thread functions were using the `void*`, so that didn't make things worse. I think we cannot get to ideal state in one step, so there will be always space for improvements. I agree the type safety is important. The new PR should improve that.